devito/builtins.py

```diff
@@ -273,6 +273,13 @@ def initialize_function(function, data, nbl, mapper=None, mode='constant',
     if isinstance(function, dv.TimeFunction):
         raise NotImplementedError("TimeFunctions are not currently supported.")
 
+    if nbl == 0:
+        if isinstance(data, dv.Function):
+            function.data[:] = data.data[:]
+        else:
+            function.data[:] = data
```

Comment:
`= data[:]`? I think without the `:` gets error in some weird case with empty dimensions

Reply:
I think it's normally fine if `data` is a numpy array but certainly doesn't hurt so will add.